ApiReview/API.Polly.Core/NoDocs/Polly/Outcome.TResult.cs

```diff
@@ -0,0 +1,15 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+
+namespace Polly;
+
+public readonly struct Outcome<TResult>
+{
+    public Exception? Exception { get; }
+    public TResult? Result { get; }
+    public void EnsureSuccess();
```

Comment:
Andrey:
I would expect Result getter to throw an exception if there was an exception.
similar to how Task.Result throws a captured exception.
but is this method even needed?

Martin:
I think we should avoid throwing exceptions from Outcome as it is low-level API used in high-perf scenarios and in delegates.

Juraj:
This is similar pattern that's used for `HttpResponseMessage` to check for successful status code. Also Cosmos DB SDK has similar implementation - you either work with DTOs directly (higher level) and get the exceptions, or you work with `ResponseMessage` (lower level) but need to check for the result explicitly using `response.EnsureSuccessStatusCode()`.

Reply:
I prefer the HttpClient-style pattern too.

Reply:
Little bit more context:

- This type is semantically/behaviorally almost identical to [`DelegateResult`](https://github.com/App-vNext/Polly/blob/282297e95b6b6ac140f1c6240fd0b21924cbb6ee/src/Polly/DelegateResult.cs#L6) or [`PolicyResult`](https://github.com/App-vNext/Polly/blob/282297e95b6b6ac140f1c6240fd0b21924cbb6ee/src/Polly/PolicyResult.cs#L6) from V7. 
- The `EnsureSuccess` was added as a feature request by user (#1238). We didn't want to add any new API to V7 so we included this method in V8.

Reply:
How do you think about renaming `EnsureSuccess` to `ThrowIfException`?

Reply:
(as I suggested this change) I believe this change will make the contract cleaner: (1) the Throw part  informs that the method can throw an exception and (2) the IfException explain the condition under which it can be thrown.